### PR TITLE
Update org.jetbrains.intellij 1.15.0 -> 1.17.0

### PR DESCRIPTION
**Pull request by mkondratek**
A few minor fixes & updates.

## Test plan
- `:runIde`, test accounts switching & rate limit errors
- plugin builds, green CI

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -23,7 +23,7 @@ plugins {
   id("java")
   // Dependencies are locked at this version to work with JDK 11 on CI.
   id("org.jetbrains.kotlin.jvm") version "1.9.10"
-  id("org.jetbrains.intellij") version "1.15.0"
+  id("org.jetbrains.intellij") version "1.17.0"
```
Comment (mkondratek) [inline build.gradle.kts:26]: I cannot see anything important for us.

https://github.com/JetBrains/gradle-intellij-plugin/releases

File: src/main/kotlin/com/sourcegraph/cody/CodyToolWindowContent.kt [hunk 2]
```diff
@@ -183,16 +181,6 @@ class CodyToolWindowContent(private val project: Project) : UpdatableChat {
     }
   }
 
-  private fun getUserId(server: CodyAgentServer): String? {
```
Comment (mkondratek) [inline src/main/kotlin/com/sourcegraph/cody/CodyToolWindowContent.kt:186]: unused

File: src/main/kotlin/com/sourcegraph/cody/agent/protocol/GetFeatureFlag.kt [hunk 3]
```diff
@@ -1,3 +1,7 @@
 package com.sourcegraph.cody.agent.protocol
 
-data class GetFeatureFlag(val flagName: String)
+data class GetFeatureFlag(val flagName: String) {
```
Comment (mkondratek) [inline src/main/kotlin/com/sourcegraph/cody/agent/protocol/GetFeatureFlag.kt:3]: I am preparing us to have two more ff here (#319). I tried to make it an enum but it did not work through the request to the agent.

File: src/main/java/com/sourcegraph/cody/SubscriptionTabUtils.kt [hunk 4]
```diff
@@ -22,8 +22,9 @@ fun fetchSubscriptionPanelData(
   if (activeAccountType != null) {
     ensureUserIdMatchInAgent(activeAccountType.id, server)
     return if (activeAccountType.isDotcomAccount()) {
-      val codyProFeatureFlag = server.evaluateFeatureFlag(GetFeatureFlag("CodyProJetBrains"))
-      if (codyProFeatureFlag.get() != null && codyProFeatureFlag.get()!!) {
+      val codyProFeatureFlag =
+          server.evaluateFeatureFlag(GetFeatureFlag.CodyProJetBrains).thenApply { it == true }.get()
```
Comment (pkukielka) [inline src/main/java/com/sourcegraph/cody/SubscriptionTabUtils.kt]: Minor nitpick, but I think this is cleaner (less code):
`server.evaluateFeatureFlag(GetFeatureFlag.CodyProJetBrains).get() == true`

And in general I would avoid `thenX` functions called on the agent server/client calls.
It's a bit hard to reason ion what context/on what thread they will be called.

**Review by pkukielka — APPROVED** [review]
LGTM

**Pull request merged**